core/dbt/contracts/graph/semantic_models.py

```diff
@@ -0,0 +1,152 @@
+from dataclasses import dataclass
```

Comment:
I am very happy about moving these out of nodes.py, but I wonder if we should move them out of the graph subdirectory?  I could see arguments either way. We don't need to decide now.